src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -26,36 +26,162 @@
 #include <ripple/protocol/JsonFields.h>
 #include <ripple/protocol/TxFlags.h>
 #include <boost/format.hpp>
-#include <boost/tokenizer.hpp>
+#include <boost/regex.hpp>
 #include <algorithm>
 #include <mutex>
 
 namespace ripple {
+
+/** Amendments that this server supports and enables by default */
+static
+Section
+preEnabledAmendments ()
+{
+    Section section ("Pre-Enabled Amendments");
+
+    return section;
+}
+
+/** Amendments that this server supports, but doesn't enable by default */
+static
+Section
+supportedAmendments ()
+{
+    Section section ("Supported Amendments");
+
+    return section;
+}
+
+static
+std::vector<std::pair<uint256, std::string>>
+parseSection (Section const& section)
+{
+    // <key> '=' <value>
+    static boost::regex const re1 (
+            "^"                    // start of line
+            "(?:\\s*)"             // whitespace (optional)
+            "([abcdefABCDEF0-9]*)" // <hex amend ID>
+            "(?:\\s*)"             // whitespace (optional)
+            "(?:.*\\S+)"           // <value>
+        , boost::regex_constants::optimize
```

Comment:
Actually, the comment is wrong. I should remove that. The expression captures any number of hex characters, and optionally, any data present after whitespace. I would consider changing the `*` in `([abcdefABCDEF0-9]*)` to a `+` or even specifying the exact number of characters needed.